simulator/infra/memory/memory.cpp

```diff
@@ -44,12 +44,21 @@ FuncMemory::FuncMemory( uint32 addr_bits,
     memory.resize(set_cnt);
 }
 
-void FuncMemory::memcpy_host_to_guest( Addr dst, const Byte* src, size_t size)
+size_t FuncMemory::memcpy_host_to_guest( Addr dst, const Byte* src, size_t size)
 {
-    for ( size_t offset = 0; offset < size; ++offset)
-        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic) Low level access
-        alloc_and_write_byte( dst + offset, *(src + offset));
-}    
+    size_t offset = 0;
+    for (; offset < size; ++offset)
```

Comment:
Clang-Tidy does not like pointer arithmetic, but here we cannot live without it.

Reply:
I assumed that Clang-Tidy won't be offended if I replace explicit pointer arithmetic `*(src+offset)` with array-styled `src[offset]`. Or will it be?

Reply:
It does not care:

```
/home/travis/build/MIPT-ILab/mipt-mips/simulator/infra/memory/memory.cpp:51:45: warning: do not use pointer arithmetic [cppcoreguidelines-pro-bounds-pointer-arithmetic]
        alloc_and_write_byte( dst + offset, src[offset]);
                                            ^
/home/travis/build/MIPT-ILab/mipt-mips/simulator/infra/memory/memory.cpp:59:9: warning: do not use pointer arithmetic [cppcoreguidelines-pro-bounds-pointer-arithmetic]
        dst[offset] = check_and_read_byte( src + offset);
        ^
```